File: app/src/main/java/com/example/instagram/reqinterface/ImageService.java
package com.example.instagram.reqinterface;

import com.example.instagram.modal.PexelResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Query;

public interface ImageService {

    @GET("v1/popular")
    Call<PexelResponse> getImages(@Header("Authorization") String api_key,
                                  @Query("page") int page,
                                  @Query("per_page") int per_page);
}

File: app/src/main/java/com/example/ktinsta/reqinterface/ImageService.kt
package com.example.ktinsta.reqinterface

import com.example.ktinsta.modal.PexelResponse
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Query

interface ImageService {
    @GET("v1/popular")
    suspend fun getImages(
        @Header("Authorization") api_key: String?,
        @Query("page") page: Int,
        @Query("per_page") per_page: Int
    ): PexelResponse
}
